src/repository.py

```diff
@@ -163,7 +163,7 @@ def _commit_file_to_tree_element(commit_file: commit_module.FileAction) -> Input
             raise NotImplementedError(f"unsupported file in commit, {commit_file}")
 
 
-class Client:
+class Client:  # pylint: disable=R0904
```

Comment:
What is this disable please?

Reply:
This class is definitely becoming to complex (hence the check disabled). We also have an issue about this [here](https://github.com/canonical/upload-charm-docs/issues/175)